ui/cypress/integration/topsql/topsql.spec.js

```diff
@@ -0,0 +1,245 @@
+      cy.get('.ant-dropdown-menu-item').eq(1).click()
+      cy.get('[data-e2e="auto-refresh-button"]').contains('30 s')
+    })
+  })
+
+  describe('Chart and table', () => {
+    it('when the time range is large, the chart interval is large', () => {
+      setCustomTimeRange('2022-01-07 00:00:00{enter}2022-01-12 00:00:00{enter}')
+      cy.get('.echCanvasRenderer').toMatchImageSnapshot()
+    })
+
+    it('when the time range is small, the chart interval is small', () => {
+      setCustomTimeRange('2022-01-12 01:00:00{enter}2022-01-12 01:01:00{enter}')
+      cy.get('.echCanvasRenderer').toMatchImageSnapshot()
+    })
+
+    it('the last item in the table list is others', () => {
+      setCustomTimeRange('2022-01-12 00:00:00{enter}2022-01-12 05:00:00{enter}')
+      cy.get('[data-e2e="topsql-list-table"] .ms-List-cell')
+        .children()
+        .eq(5)
+        .find('[data-e2e="topsql-listtable-row-others"]')
+    })
+
+    it('table has top 5 records and the others record', () => {
+      setCustomTimeRange('2022-01-12 00:00:00{enter}2022-01-12 05:00:00{enter}')
+
+      cy.get('[data-e2e="topsql-list-table"] .ms-List-cell')
+        .children()
+        .should('have.length', 6)
+
+      cy.get('[data-e2e="topsql-list-table"] .ms-List-cell').each((item) => {
+        cy.wrap(item).trigger('mouseover')
+        cy.get('.echCanvasRenderer').toMatchImageSnapshot()
+      })
+    })
+
+    it('table can only be single selected', () => {
+      setCustomTimeRange('2022-01-12 00:00:00{enter}2022-01-12 05:00:00{enter}')
+
+      cy.get('[data-e2e="topsql-list-table"] .ms-List-cell').each((item) => {
+        cy.wrap(item).click()
+        cy.get(
+          '[data-e2e="topsql-list-table"] .ms-DetailsRow-check[aria-checked="true"]'
+        ).should('have.length', 1)
+      })
+    })
+  })
+
+  describe('Top SQL settings', () => {
+    it('close Top SQL by settings panel, the chart and table will still work', () => {
+      cy.get('[data-e2e="topsql-settings"]').click()
+      cy.wait('@getTopsqlSettings')
+
+      cy.get('[data-e2e="topsql-settings-enable"]').click()
+      cy.get('[data-e2e="topsql-settings-save"]').click()
+      cy.get('.ant-btn-primary.ant-btn-dangerous').click()
+      cy.get('[data-e2e="topsql-not-enabled-alert"]').should('exist')
+
+      setCustomTimeRange('2022-01-12 00:00:00{enter}2022-01-12 05:00:00{enter}')
+
+      enableTopSQL()
+      cy.wait('@getTopsqlSettings')
+      cy.get('[data-e2e="topsql-not-enabled-alert"]').should('not.exist')
+    })
+  })
+
+  describe('SQL statement details', () => {
+    it('click one table row, show the list detail table and information contents', () => {
+      setCustomTimeRange('2022-01-12 00:00:00{enter}2022-01-12 05:00:00{enter}')
+
+      cy.get('[data-e2e="topsql-list-table"] .ms-List-cell').eq(0).click()
+      cy.get('[data-e2e="topsql-listdetail-table"]').should('exist')
+      cy.get('.e2e-topsql-listdetail-content-sql_text').should('exist')
```

Comment:
I see that some are using classes and some are using attributes. Are there ways to make them unified?

Reply:
Let me try.